### PR TITLE
DEV-3824: Re-enable swag element

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

- Re-enables swag elements in contribution pages and updates the editor panel to match new designs.
- Updates the element as displayed in a contribution page to send the correct encoded value for `swag_choices`.
- Replaces `validateInputPositiveFloat` with `parseFloatStrictly`.

#### Why are we doing this? How does it help us?

- Unblocks rev engine migration for some clients.
- re: `parseFloatStrictly`, I needed to allow the swag contribution threshold to be set to 0 but otherwise filter out non-numeric values. Refactoring to just parse the number meant we didn't have to create a whole other function for this.

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

Yes, we re-enabled the swag element.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

Should reach out to clients who would make use of this.

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3824

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

**Review by GuiMend — APPROVED** [review]
LGTM @nrh-cklimas

**Review by GuiMend — CHANGES_REQUESTED** [review]
1) Breaking page
image
When trying to edit "Contributor Benefits" the page breaks. (Testing with this donation page: https://dev-3824.revengine-review.org/edit/pages/54/ ) 
Btw: I removed and re-added the block and the error stopped occurring. (Might be an outdated config issue, but I didn't dig into too much.)

2) Updates not showing on preview:
After making changes to the element and clicking "update" nothing happens:
https://dev-3824.revengine-review.org/edit/pages/67/
(Even after saving the page it still doesn't show the changes)
image

3) Validation is working, but max chars doesn't show error, it only stops the user from entering more chars
image
Also, perhaps make the text break line and wrap so that we don't get the scrollbar?
image

**Comment by nrh-cklimas** [comment]
@GuiMend thanks for the review!

1. I looked at that page and the element content is totally empty. What I'd like to do is since we have very few if any pre-existing swag elements in prod, would like to address individually instead of writing conditions around the element missing all data. It's something I'll flag in QA though.

2. That's a side effect of the swag threshold--if you change the contribution amount in the page preview to hit the threshold, you should see the options.

3. Talked to Rachel and she is OK with keeping the field max length as our validation method here. I did add a fix for the scrolling caused by long option names in 2bdddc9982f27d191ac29528cba508b30120baf7.

**Review by GuiMend — APPROVED** [review]
👍

File: spa/src/components/pageEditor/editInterface/pageElements/addElementModal/AddElementModal.tsx [hunk 1]
```diff
@@ -49,29 +52,15 @@ function AddElementModal({ addElementModalOpen, setAddElementModalOpen, destinat
...
-  //     Object.values(dynamicLayoutElements as ElementRecord).filter(({ type }) =>
-  //       (page?.plan?.page_elements ?? []).includes(type)
-  //     )
-  //   );
-  //   setPermittedSidebarElements(
-  //     Object.values({ ...dynamicSidebarElements } as SidebarElementRecord).filter(({ type }) =>
-  //       (page?.plan?.sidebar_elements ?? []).includes(type)
-  //     )
-  //   );
-  // }, [page?.plan?.page_elements, page?.plan?.sidebar_elements]);
```
Comment (benjaminEwhite) [inline spa/src/components/pageEditor/editInterface/pageElements/addElementModal/AddElementModal.tsx:65]: Get rid of this?

**Review by benjaminEwhite — COMMENTED** [review]
Saw one place that there's some commented code that could be removed.

File: spa/src/components/pageEditor/editInterface/pageElements/addElementModal/AddElementModal.tsx (hunk 1 above)
Comment (nrh-cklimas) [reply spa/src/components/pageEditor/editInterface/pageElements/addElementModal/AddElementModal.tsx:65]: I did? :)

**Pull request merged**